components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/impl/AuthenticatorManagementDAOImpl.java

```diff
@@ -0,0 +1,340 @@
+        try (NamedPreparedStatement statement = new NamedPreparedStatement(dbConnection, Query.GET_AUTHENTICATOR_SQL)) {
+
+            statement.setString(Column.NAME, authenticatorConfigName);
+            statement.setInt(Column.TENANT_ID, tenantId);
+
+            try (ResultSet rs = statement.executeQuery()) {
+                if (rs.next()) {
+                    if (AuthenticationType.VERIFICATION.toString().equals(rs.getString(Column.AUTHENTICATION_TYPE))) {
+                        config = new VerificationAuthenticatorConfig();
+                    } else {
+                        config = new LocalAuthenticatorConfig();
+                    }
+                    config.setName(rs.getString(Column.NAME));
+                    config.setDisplayName(rs.getString(Column.DISPLAY_NAME));
+                    config.setEnabled(rs.getString(Column.IS_ENABLED).equals(IS_TRUE_VALUE));
+                    config.setDefinedByType(DefinedByType.valueOf(rs.getString(Column.DEFINED_BY)));
+                }
+            }
+
+            if (config == null) {
+                return null;
+            }
+
+            int configId = getAuthenticatorIdentifier(dbConnection, config.getName(), tenantId);
+            try (NamedPreparedStatement statementProp = new NamedPreparedStatement(dbConnection,
+                    Query.GET_AUTHENTICATOR_PROP_SQL)) {
+
+                statementProp.setInt(Column.AUTHENTICATOR_ID, configId);
+                statementProp.setInt(Column.TENANT_ID, tenantId);
+
+                try (ResultSet rs = statementProp.executeQuery()) {
+                    List<Property> properties = new ArrayList<>();
+                    while (rs.next()) {
+                        Property property = new Property();
+                        property.setName(rs.getString(Column.PROPERTY_KEY));
+                        property.setValue(rs.getString(Column.PROPERTY_VALUE));
+                        property.setConfidential(Boolean.parseBoolean(rs.getString(Column.IS_SECRET)));
+                        properties.add(property);
+                    }
+                    config.setProperties(properties.toArray(new Property[0]));
+                }
+            }
+
+            IdentityDatabaseUtil.commitTransaction(dbConnection);
+            return config;
+        } catch (SQLException e) {
+            ErrorMessages error = ErrorMessages.ERROR_WHILE_RETRIEVING_AUTHENTICATOR_BY_NAME;
+            throw new AuthenticatorMgtServerException(error.getMessage(), error.getMessage(), error.getCode(), e);
+        }
+    }
+
+    private boolean isBasicInfoUpdated(LocalAuthenticatorConfig existingAuthenticatorConfig,
+                                       LocalAuthenticatorConfig updatedAuthenticatorConfig) {
+
+        return !existingAuthenticatorConfig.getDisplayName().equals(updatedAuthenticatorConfig.getDisplayName()) ||
+                existingAuthenticatorConfig.isEnabled() != updatedAuthenticatorConfig.isEnabled();
+    }
```

Comment:
don't quite understand the use of this. why we really need this check ?

Reply:
Addressed with commit d72ebd31b7150e7c562faf83ee80300a60e1e5d8